lib/symengine/basic.rb

```diff
@@ -11,8 +11,13 @@ def free_symbols
     end
     
     def abs
-        return SymEngine::abs(self)
+      SymEngine::abs(self)
+    end
+    def to_proc()
+      if self.free_symbols.count > 1
+        raise ArgumentError, "Too many free symbols! Only 1 allowed. Found #{self.free_symbols.count}."
```

Comment:
Overall, I think for the ruby wrappers, lot more 'rubification' is needed. For example, to support `.to_proc(:x, :y)` we'd need to map Ruby symbols to SymEngine symbols. Until then I can implement `.to_proc(x, y)` where `x` and `y` are of `SymEngine::Symbol` type. Will add this as an issue too.